napari/utils/events/dataclass.py

```diff
@@ -478,4 +527,6 @@ def evented_dataclass(
         # convert public dataclass fields to properties
         cls = convert_fields_to_properties(cls)
     setattr(cls, '_get_state', _dc.asdict)
+    setattr(cls, 'as_dict', _dc.asdict)
+    setattr(cls, 'update_from_dict', update_from_dict)
```

Comment:
Curious what people think of these names `as_dict` and `update_from_dict`. Alternatives could be just `from_dict` or `asdict` and `fromdict` (to match the dataclass syntax)?

I like `as_dict`. `update_from_dict` might be a bit long, but at least it is clear, so I'm ok with it too. I imagine we'll use these quite a lot though so good to talk it over @jni

Reply:
I do kind of wish that `dataclasses` had used `as_dict` instead of `asdict` in the first place, but now that they _have_ chosen `asdict`, I'm slightly inclined to stick with the standard library phrasing.

`fromdict` has sort of changed here, since as @Czaki pointed out, "from_dict" can be done just by `Class(**dict)`, and what's been implemented here really is more of an `update`.  It'd be nice to use `update` alone (following `dict.update`?), but since this will be on _every_ `evented_dataclass`, I don't think we want to use up that API.

Reply:
> It'd be nice to use update alone (following dict.update?), but since this will be on every evented_dataclass, I don't think we want to use up that API.

I don't know, I think it might be ok - it's an update, like `dict.update`, I kinda like it. I'd be supportive of `asdict` and `update` as the two names

Reply:
I especially use the longer name to avoid name, collition. But `update` should be fine if it could have dict, or instance of the class as an argument?  

For me `update` look like a method that will accept class instance.

Reply:
> But update should be fine if it could have dict, or instance of the class as an argument?

Yeah I think it should be able to accept both if we go in this direction

Reply:
updated